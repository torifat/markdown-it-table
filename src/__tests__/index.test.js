import fs from 'fs';
import path from 'path';

import markdownIt from 'markdown-it';
import markdownItTable from '../';

const getMDFromFixture = name => {
  const filename = path.resolve(__dirname, `__fixtures__/${name}.md`);
  return fs.readFileSync(filename, 'utf8');
};

describe('markdown-it-table', () => {
  let md;
  beforeAll(() => {
    md = markdownIt('commonmark').use(markdownItTable, {});
  });

  it('should be able to parse simple table', () => {
    const input = getMDFromFixture('simple');
    // const out = md.parse(input);
    expect(md.render(input)).toMatchSnapshot();
  });

  it('should be able to parse cell with blockquote', () => {
    const input = getMDFromFixture('blockquote');
    expect(md.render(input)).toMatchSnapshot();
  });

  it.skip('should be able to parse table with redundant indentations', () => {
    const input = getMDFromFixture('indentation');
    expect(md.render(input)).toMatchSnapshot();
  });
});
