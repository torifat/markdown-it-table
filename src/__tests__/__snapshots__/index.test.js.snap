// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`markdown-it-table should be able to parse cell with blockquote 1`] = `
"<table>
<tr>
<th>
<p>Heading 1</p>
</th>
<th>
<p>Heading 2</p>
</th>
</tr>
<tr>
<td>
<p>Cell 1</p>
</td>
<td>
<blockquote>
<p>Blockquote 1</p>
</blockquote>
</td>
</tr>
<tr>
<td>
<blockquote>
<p>Blockquote 2</p>
</blockquote>
</td>
<td>
<p>Cell 4</p>
</td>
</tr>
</table>
"
`;

exports[`markdown-it-table should be able to parse simple table 1`] = `
"<table>
<tr>
<th>
<p>Heading 1</p>
</th>
<th>
<p>Heading 2</p>
</th>
</tr>
<tr>
<td>
<p>Cell 1</p>
</td>
<td>
<p>Cell 2</p>
</td>
</tr>
<tr>
<td>
<p>Cell 3</p>
</td>
<td>
<p>Cell 4</p>
</td>
</tr>
</table>
"
`;

exports[`markdown-it-table should be able to parse table with redundant indentations 1`] = `
"<table>
<tr>
<th>
<p>Heading 1</p>
</th>
<th>
<p>Heading 2</p>
</th>
</tr>
<tr>
<td>
<p>Cell 1</p>
</td>
<td>
<p>Cell 2</p>
</td>
</tr>
<tr>
<td>
<p>Cell 3</p>
</td>
<td>
<p>Cell 4</p>
</td>
</tr>
</table>
"
`;
